api/grpc_server.go

```diff
@@ -321,25 +322,41 @@ func (s SpacemeshGrpcService) SetLoggerLevel(ctx context.Context, msg *pb.SetLog
 
 func (s SpacemeshGrpcService) GetAccountTxs(ctx context.Context, txsSinceLayer *pb.GetTxsSinceLayer) (*pb.AccountTxs, error) {
 	log.Info("GRPC GetAccountTxs msg")
-	acc := types.HexToAddress(txsSinceLayer.Account.Address)
-	if txsSinceLayer.StartLayer > uint64(s.Tx.LatestLayer()) {
+
+	currentPBase := s.Tx.ValidatedLayer()
+
+	addr := types.HexToAddress(txsSinceLayer.Account.Address)
+	minLayer := types.LayerID(txsSinceLayer.StartLayer)
+	if minLayer > s.Tx.LatestLayer() {
 		return &pb.AccountTxs{}, fmt.Errorf("invalid start layer")
 	}
-	var allTxs []types.TransactionId
-	for i := txsSinceLayer.StartLayer; i < uint64(s.Tx.LatestLayer()); i++ {
-		txs := s.Tx.GetTransactionsByDestination(types.LayerID(i), acc)
-		allTxs = append(allTxs, txs...)
-		moreTxs := s.Tx.GetTransactionsByOrigin(types.LayerID(i), acc)
-		allTxs = append(allTxs, moreTxs...)
+
+	txs := pb.AccountTxs{ValidatedLayer: currentPBase.Uint64()}
+
+	blockTxIds := s.getBlockTxIds(minLayer, addr)
+	for _, txId := range blockTxIds {
+		txs.Txs = append(txs.Txs, txId.String())
 	}
 
-	txs := pb.AccountTxs{}
-	for _, x := range allTxs {
-		txs.Txs = append(txs.Txs, x.String())
+	mempoolTxIds := s.TxMempool.GetTxIdsByAddress(addr)
+	for _, txId := range mempoolTxIds {
+		txs.Txs = append(txs.Txs, txId.String())
 	}
+
 	return &txs, nil
 }
 
+func (s SpacemeshGrpcService) getBlockTxIds(minLayer types.LayerID, addr types.Address) []types.TransactionId {
```

Comment:
this func doesnt concern blocks so perhaps you can find a better name

Reply:
Got this feedback from @barakshani already, totally agree, I changed it to `getTxIdsFromMesh`.